client/ayon_core/pipeline/context_tools.py

```diff
@@ -579,3 +579,17 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def version_up_workfile():
+    """Function to increment and save workfile
+    """
+    host = registered_host()
+    current_file = host.get_current_workfile()
+    if not current_file:
+        log.debug(
+            "Current file is unsaved and can't be versioned up. "
+            "Please save your file first.")
+        return None
+    filepath = version_up(current_file)
```

Comment:
This will version up current file into next version, which might potentially override existing file?

Reply:
> This will version up current file into next version, which might potentially override existing file?

Looking at the logic it actually checks for existing files and versions up beyond that?